server/common/utils/type_conversion_utils.py

```diff
@@ -117,23 +122,19 @@ def _get_type_info(array: Union[np.ndarray, pd.Series, pd.Index]) -> Tuple[np.dt
 
     dtype = array.dtype
 
-    res = _get_type_info_from_dtype(dtype)
-    if res is not None:
-        return res
+    if not isinstance(array, sp.spmatrix):
+        res = _get_type_info_from_dtype(dtype)
+        if res is not None:
+            return res
 
+    if isinstance(array, sp.spmatrix):
+        return ("sparse" + str(dtype.itemsize * 8), {"type": dtype.name})
     if dtype.kind == "O":
         if dtype.name == "category":
-            # Sometimes CategoricalDType can be encoded as int or float without further fuss.
-            # Do not specify the categories in the schema - let the client-side figure it out
-            # on its own.  Utilize Series.to_numpy() to do casting that handles categorical
-            # NA/NaN (missing or undefined) categories.
-            if dtype.categories.dtype.kind in ["f", "i", "u"]:
-                return (
-                    _get_type_info(array.to_numpy())[0],
-                    {"type": "categorical"},
-                )
-            else:
-                return (np.dtype(str), {"type": "categorical", "categories": dtype.categories.to_list()})
+            return (
+                "cat" + str(array.cat.codes.dtype.itemsize * 8),
```

Comment:
This violates the signature for the function as "cat8"  is not a numpy.dtype. Feels hackish and breaks the abstraction.

One possible solution is to return a real dtype, ie, follow the Pandas lead and return a  pd.CategoricalDtype (which contains the type of the underlying category data).

Reply:
I don't think pd.CategoricalDtype contains the type of the code array.

Reply:
Try:
* for codes:    series.cat.codes.dtype
* for the lables/categories:   series.cat.categories.dtype